projects/m5/001-budget-app/js/index.js

```diff
@@ -0,0 +1,225 @@
+//m5-001-budget-app
+
+// Class that create a budjet category with the desired name
+class Category {
+    constructor(name) {
+        //arguments
+        this.name = name;
+        this.ledger = [];
+    }
+
+    //methods
+    /**
+     * deposit allows to add an obj to the ledger with the form of {'amount': quantity, 'description':description} 
+     * @param {number} quantity the max digit of the number is 7 
+     * @param {string} description it can also be omitted
+     * @returns add an obj to the ledger list
+     */
+    deposit(quantity, description){
+        if (quantity > 99999.99){
+                throw new Error(`Deposit amount can not contain more than 7 digits.
+                The following deposit has blocked the process: amount = ${quantity}, descritpion = ${description}`)
+        }else{
+            description = description || ''; // same as: if(description===undefined){description=''}
+            return this.ledger.push({'amount': quantity, 'description':description})
```

Comment:
This method directly modifies the instance variable this.ledger .... so you don't need to return a value ;-)